tests/framework/test_engine.py

```diff
@@ -172,6 +172,39 @@ def test_SimulationContext_name_management(SimulationContext):
     }
 
 
+def test_SimulationContext_run(SimulationContext, mocker):
+    sim = SimulationContext()
+
+    # Mock the methods called by sim.run()
+    sim.setup = mocker.Mock()
+    sim.initialize_simulants = mocker.Mock()
+    sim.run = mocker.Mock()
+    sim.finalize = mocker.Mock()
+    sim.report = mocker.Mock()
+
+    sim.run_simulation()
+
+    # Assert each mocked method was called as expected
+    sim.setup.assert_has_calls([mocker.call()], any_order=True)
+    sim.initialize_simulants.assert_has_calls([mocker.call()], any_order=True)
+    sim.run.assert_has_calls([mocker.call()], any_order=True)
+    sim.finalize.assert_has_calls([mocker.call()], any_order=True)
+    sim.report.assert_has_calls([mocker.call()], any_order=True)
```

Comment:
Don't we care about the order? We do have a test somewhere in pseudopeople that tests the order of the application of noise functions if you want to test it that closely.

Reply:
I figured order was maintained by vivarium itself but yeah, I'll go remind myself how I did order checks in psp

Reply:
We do, but I figured that's enforced by vivarium and I recalled it being somewhat difficult to implement in psp. I'll have a look though